app/Cardano/Shell/Features/Logging.hs

```diff
@@ -68,44 +71,48 @@ createLoggingFeature cardanoEnvironment cardanoConfiguration = do
     -- the filesystem, so we give him the most flexible/powerful context, @IO@.
     loggingConfiguration    <-  LoggingParameters <$> (Config.setup $ ccLogConfig cardanoConfiguration)
 
+    (baseTrace, switchBoard) <- setupTrace_ (lpConfiguration loggingConfiguration) "cardano"
+
     -- we construct the layer
-    loggingLayer            <- (featureInit loggingCardanoFeatureInit)
+    loggingLayer            <- (featureInit $ loggingCardanoFeatureInit loggingConfiguration baseTrace switchBoard)
                                cardanoEnvironment
                                NoDependency
                                cardanoConfiguration
                                loggingConfiguration
 
     -- we construct the cardano feature
-    let cardanoFeature      = loggingCardanoFeature loggingCardanoFeatureInit loggingLayer
+    let cardanoFeature      = loggingCardanoFeature (loggingCardanoFeatureInit loggingConfiguration baseTrace switchBoard) loggingLayer
 
     -- we return both
     pure (loggingLayer, cardanoFeature)
 
-loggingCardanoFeatureInit :: LoggingCardanoFeature
-loggingCardanoFeatureInit = CardanoFeatureInit
-    { featureType    = "LoggingMonitoringFeature"
-    , featureInit    = initLogging
-    , featureCleanup = cleanupLogging
-    }
-  where
-    initLogging :: CardanoEnvironment -> NoDependency -> CardanoConfiguration -> LoggingParameters -> IO LoggingLayer
-    initLogging _ _ _ loggingParameters = do
-        baseTrace <- setupTrace (Right (lpConfiguration loggingParameters)) "cardano"
-        pure $ LoggingLayer
-                { llBasicTrace  = Trace.natTrace liftIO baseTrace
-                , llLogDebug    = Trace.logDebug
-                , llLogInfo     = Trace.logInfo
-                , llLogNotice   = Trace.logNotice
-                , llLogWarning  = Trace.logWarning
-                , llLogError    = Trace.logError
-                , llAppendName  = Trace.appendName
-                }
-    cleanupLogging :: LoggingLayer -> IO ()
-    cleanupLogging loggingLayer = shutdownTrace $ llBasicTrace loggingLayer
+loggingCardanoFeatureInit :: LoggingParameters -> Trace IO Text -> Switchboard Text -> LoggingCardanoFeature
+loggingCardanoFeatureInit _ baseTrace switchBoard = do
```

Comment:
Is the `Switchboard` part of the logging or the monitoring part?

Reply:
`Switchboard` dispatches the log messages to the selected backends.

Reply:
Thanks. It's good then.